internal/cmd/backup.go

```diff
@@ -122,8 +123,31 @@ func runBackup(c *cli.Context) error {
 	if !c.Bool("exclude-repos") && !c.Bool("database-only") {
 		reposDump := filepath.Join(rootDir, "repositories.zip")
 		log.Info("Dumping repositories in %q", conf.Repository.Root)
-		if err = zip.PackTo(conf.Repository.Root, reposDump, true); err != nil {
-			log.Fatal("Failed to dump repositories: %v", err)
+		if c.Bool("exclude-mirror-repos") {
+			repos, err := db.GetRepositoriesNonMirror()
+			if err != nil {
+				log.Fatal("Mirror repositories failed to load %v", err)
+			}
+			repoZip, err := zip.Create(reposDump)
+			if err != nil {
+				log.Fatal("Failed to create 'repositories.zip': %v", err)
+			}
+			baseRepoPath := filepath.Base(conf.Repository.Root)
+			repoZip.AddEmptyDir(baseRepoPath)
+			for _, r := range repos {
+				name := filepath.Join(r.Owner.Name, fmt.Sprintf("%s%s", r.Name, ".git"))
```

Comment:
It looks like an overkill to use `fmt.Sprintf`:
```suggestion
				name := filepath.Join(r.Owner.Name, r.Name + ".git")
```